src/javascript/app_2/App/Components/Elements/InkBar/with_inkbar.jsx

```diff
@@ -0,0 +1,89 @@
+import classnames from 'classnames';
+import PropTypes  from 'prop-types';
+import React      from 'react';
+import ReactDOM   from 'react-dom';
+import { InkBar } from './inkbar.jsx';
+
+const withInkBar = (Component) => {
+    class ComponentWithInkBar extends React.Component {
+        state = {
+            left : 0,
+            width: 0,
+        };
+
+        componentDidMount() {
+            this.node = ReactDOM.findDOMNode(this); // eslint-disable-line
+            this.updateInkbarPosition(this.node.querySelector('a[class="active"]'));
+        }
+
+        componentDidUpdate() {
+            const active_el = this.node.querySelector('a[class="active"]');
+            if (active_el) {
+                this.updateInkbarPosition(active_el);
+            } else if (this.state.left !== 0 || this.state.width !== 0) {
+                this.clearInkBar(); // clear InkBar when active element doesn't exist
+            }
+        }
+
+        componentWillMount() {
```

Comment:
@kellybinary 
this lifecycle method is deprecated in the latest version of React, it's better to use the `constructor` instead.

Reply:
sure, will do 👍